pallets/system/src/header_builder.rs

```diff
@@ -85,186 +85,24 @@ pub trait HeaderExtensionBuilder {
-
-	match version {
-		HeaderVersion::V3 => {
-			use avail_core::kate_commitment::v3::KateCommitment;
-			let kate = KateCommitment::new(1, 4, data_root, empty_commitment);
-			v3::HeaderExtension {
-				app_lookup: empty_app_lookup,
-				commitment: kate,
-			}
-			.into()
-		},
-	}
-}
-
-#[cfg(feature = "std")]
-pub fn build_extension(
-	app_extrinsics: &[AppExtrinsic],
-	data_root: H256,
-	block_length: BlockLength,
-	_block_number: u32,
-	seed: Seed,
-	version: HeaderVersion,
-) -> HeaderExtension {
-	use avail_base::metrics::avail::HeaderExtensionBuilderMetrics;
-	use avail_core::header::extension::v3;
-
-	let build_extension_start = std::time::Instant::now();
-
-	// Build the grid
-	let timer = std::time::Instant::now();
-	let maybe_grid = build_grid(app_extrinsics, block_length, seed);
-	// Evaluation Grid Build Time Metrics
-	HeaderExtensionBuilderMetrics::observe_evaluation_grid_build_time(timer.elapsed());
-	let grid = match maybe_grid {
-		Ok(res) => res,
-		Err(message) => {
-			log::error!("NODE_CRITICAL_ERROR_001 - A critical error has occurred: {message:?}.");
-			log::error!("NODE_CRITICAL_ERROR_001 - If you see this, please warn Avail team and raise an issue.");
-			HeaderExtensionBuilderMetrics::observe_total_execution_time(
-				build_extension_start.elapsed(),
-			);
-			return get_empty_header(data_root, version);
-		},
-	};
-
-	// Build the commitment
-	let timer = std::time::Instant::now();
-	let maybe_commitment = build_commitment(&grid);
-	// Commitment Build Time Metrics
-	HeaderExtensionBuilderMetrics::observe_commitment_build_time(timer.elapsed());
-	let commitment = match maybe_commitment {
-		Ok(res) => res,
-		Err(message) => {
-			log::error!("NODE_CRITICAL_ERROR_002 - A critical error has occurred: {message:?}.");
-			log::error!("NODE_CRITICAL_ERROR_002 - If you see this, please warn Avail team and raise an issue.");
-			HeaderExtensionBuilderMetrics::observe_total_execution_time(
-				build_extension_start.elapsed(),
-			);
-			return get_empty_header(data_root, version);
-		},
-	};
-
-	// Note that this uses the original dims, _not the extended ones_
-	let rows = grid.dims().rows().get();
-	let cols = grid.dims().cols().get();
-
-	// Grid Metrics
-	HeaderExtensionBuilderMetrics::observe_grid_rows(rows as f64);
-	HeaderExtensionBuilderMetrics::observe_grid_cols(cols as f64);
-
-	let app_lookup = grid.lookup().clone();
-
-	let header_extension = match version {
-		HeaderVersion::V3 => {
-			use avail_core::kate_commitment::v3::KateCommitment;
-			let kate = KateCommitment::new(rows, cols, data_root, commitment);
-			v3::HeaderExtension {
-				app_lookup,
-				commitment: kate,
-			}
-			.into()
-		},
-	};
-
-	// Total Execution Time Metrics
-	HeaderExtensionBuilderMetrics::observe_total_execution_time(build_extension_start.elapsed());
-
-	header_extension
-}
-
 /// Hosted function to build the header using `kate` commitments.
 #[runtime_interface]
 pub trait HostedHeaderBuilder {
+	#[version(1)]
```

Comment:
Since the `version` parameter is removed from the hosted function, adding a note emphasizing the necessity of creating a new HostedFunction version for every new header_version, while retaining the older ones, might be helpful.